clang/lib/Headers/float.h

```diff
@@ -7,13 +7,21 @@
  *===-----------------------------------------------------------------------===
  */
 
-#ifndef __CLANG_FLOAT_H
-#define __CLANG_FLOAT_H
-
 #if defined(__MVS__) && __has_include_next(<float.h>)
+#include <__float_header_macro.h>
```

Comment:
Previously, the macro was always defined, now it's only being defined for `__MVS__`, is that intentional?

I took a look on SourceGraph and it doesn't look like anyone is testing `__CLANG_FLOAT_H`, but it's possible someone is relying on that to determine whether float.h was included for Clang or not.